codec/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java

```diff
@@ -206,7 +206,7 @@ protected ByteBuf internalBuffer() {
     }
 
     @Override
-    public final void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
+    public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
```

Comment:
The issue is the buffer may be released by this method prematurely due to re-entry type issues. We have a few options on how to approach this but child classes not having the opportunity to control when the buffer released is a problem. We could deprecate `handlerRemoved0`, or do you have a more preferred approach?